app/data_handler.py

```diff
@@ -21,12 +21,17 @@ def addHostToDB(self, hostname, ipv4_addr, type, ios_type, local_creds):
         """Add host to database.  Returns True if successful."""
         try:
             host = app.models.Host(hostname=hostname, ipv4_addr=ipv4_addr,
-                                   type=type, ios_type=ios_type,
+                                   type=type.capitalize(),
+                                   ios_type=ios_type,
                                    local_creds=local_creds)
             app.db.session.add(host)
             # This enables pulling ID for newly inserted host
             app.db.session.flush()
             app.db.session.commit()
+        except (IntegrityError, InvalidRequestError) as e:
+            return False, 0, e
```

Comment:
should there be an app.db.session.rollback() here?

Reply:
Yes.  Added